web/pages/api/v0/dream.ts

```diff
@@ -0,0 +1,104 @@
+import { getDownloadURL, ref, uploadBytesResumable } from 'firebase/storage'
+import { nanoid } from 'nanoid'
+import { NextApiRequest, NextApiResponse } from 'next'
+import { generateAsync } from 'stability-client'
+import { storage } from 'web/lib/firebase/init'
+import {
+  CORS_ORIGIN_MANIFOLD,
+  CORS_ORIGIN_LOCALHOST,
+} from 'common/envs/constants'
+import { applyCorsHeaders } from 'web/lib/api/cors'
+
+export const config = { api: { bodyParser: true } }
+
+// Highly experimental. Proxy for https://github.com/vpzomtrrfrt/stability-client
+export default async function route(req: NextApiRequest, res: NextApiResponse) {
```

Comment:
@mqp Was wondering if I could get your help with configuring this API -- for some reason, the way I have this coded, it doesn't work when I pass in the header `Content-Type: application/json`, but if I leave it as `Content-Type: plain/text` everything parses correctly....

Am I doing something wrong?

Reply:
Well, it's weird that you both have `bodyParser: true`, and you then also call `JSON.parse(req.body)`. Since it seems like you only want to accept JSON, So I would assume that `JSON.parse(req.body)` is going to fail if you call it with `application/json` because it will already be an object instead of a string.

I would just use the `bodyParser` and then complain in the endpoint if the body is something other than an object with the fields you want.

Reply:
Doh, that was the issue  -- using `JSON.parse(req.body)`. Thanks!